### DISCUSSION TITLE
Road to 0.74.0

### DISCUSSION THREAD

**Discussion by huntie** [reactwg/react-native-releases#104] [Releases]
# Release Coordination for 0.74 is moving to Github Projects

# Now locked -> please submit pick requests and issue reports via https://github.com/reactwg/react-native-releases/issues/new/choose

All pick requests should now be filed through the [issue template](https://github.com/reactwg/react-native-releases/issues/new?assignees=&labels=Type%3A+Pick+Request&projects=&template=pick_request_form.yml)

You can see outstanding pick requests for 0.74 in its [dedicated project board](https://github.com/orgs/reactwg/projects/5/views/3)

See https://github.com/reactwg/react-native-releases/discussions/117 for more details

We will lock this discussion for now.

--- 
## Notice

- Current release candidate: [0.74.0-rc.0](https://github.com/facebook/react-native/releases/tag/0.74.0-rc.0)
- Have an issue with current release candidate? [File an issue](https://github.com/facebook/react-native/issues/new?assignees=&labels=Needs%3A+Triage+%3Amag%3A%2CType%3A+Upgrade+Issue&template=upgrade_regression_form.yml) and we will triage.
- Have a pick request for this release? Does it fall under our [pick request qualifications](https://reactnative.dev/contributing/release-faq)? If so please create a PR against the release branch and comment with the PR link
- If you are release testing, copy and fill a [test checklist](https://reactnative.dev/contributing/release-testing#testing-pre-releases-rc-on-production-apps).
- Are you testing nightlies? Please report in the dedicated discussion here: https://github.com/reactwg/react-native-releases/discussions/49

### 0.74 Release Crew 👋🏻

- Alex Hunt (Meta) @huntie 
- Luna Wei (Meta) @lunaleaps (will be out after March 8, 2024)
- Nicola Corti (Meta) @cortinico 
- Hur Ali (Callstack) @hurali97
- Alan Hughes (Expo) @alanjhughes
- Alfonso Curbelo (Coinbase) @alfonsocj

### Pick requests (this thread)

During the ~1 month **RC phase**, we will accept hotfixes only. Commits on `main` after this date will be made towards the following release.

<details>
<summary>

## Key changes in this release

</summary>

> This list includes significant changes (flagged by the Release Crew) in React Native 0.74 which have been committed to `main` ahead of the branch cut. The full set of changes will be represented in the release changelog.

### Breaking changes

- [Incoming]

### Infra

- Yarn 3 is now the default package manager when using `react-native init` https://github.com/react-native-community/cli/pull/2134

</details>

### Release Checklist

- [ ] Changelog PR
- [ ] Start a draft doc of blog post for release
- [ ] Follow up on release dependencies
- [ ] Update changelog and merge all changelogs for RCs into 0.74 heading
- [ ] Ship blog post 
- [ ] Prepare typescript DT repo PR
- [ ] Make PR to `react-native-website` with the new version cut ([see docs](https://github.com/facebook/react-native-website#cutting-a-new-version))

**Comment by cortinico** [comment]
Can we pick https://github.com/facebook/react-native/commit/4ecf57ead2a32294cb9ed19088e7cce59702794e
This is the TurboModule C++ Autolinking. It missed the branch cut by 1 hour or so.

**Reply by huntie** [reply]
Picked ✅

**Comment by coofzilla** [comment]
> During the ~1 month RC phase, we will accept hotfixes only. Commits on main after this date will be made towards the following release.

I understand this; but, is there any word on integrating `network inspection` with the hermes debugger?

**Reply by huntie** [reply]
This won't happen for 0.74 — we are currently reimplementing React Native's debugger server (an in-place replacement of the Hermes debugger), and will then be building new Network inspection support on top of this. At this time, we aren't committing to a public timeline.

https://github.com/react-native-community/discussions-and-proposals/discussions/733#discussioncomment-8557269

**Comment by Emmy-Akintz** [comment]
I'm new here

Hello!

A MERN fullstack web developer...
Emmanuel by name
You can refer to me as Emmy

**Comment by D4ryl00** [comment]
Can we pick - https://github.com/facebook/react-native/issues/34400 ?

**Reply by efstathiosntonas** [reply]
@D4ryl00 you probably mean this PR: https://github.com/facebook/react-native/commit/bab9c839eaad1ffc358fa5c87440a6de7cbe981b

**Reply by D4ryl00** [reply]
Yes this is this PR, thanks!

**Reply by kelset** [reply]
it's already in RC0

**Comment by okwasniewski** [comment]
Hey, can we pick https://github.com/facebook/react-native/pull/43146 ?

**Reply by cortinico** [reply]
This will most likely land in RC2 as it's not reviewed yet internally (cc @dmytrorykun)

**Comment by cortinico** [comment]
Please pick a CLI bump that should include this:
- https://github.com/react-native-community/cli/pull/2306

As autolinking is currently broken on New Architecture (see https://github.com/facebook/react-native/issues/43145)

**Reply by cortinico** [reply]
Commit to pick is here: https://github.com/facebook/react-native/pull/43153

**Comment by philIip** [comment]
hi can we pick https://github.com/facebook/react-native/pull/43054? it's alternative support for C++ TurboModule autolinking that barely missed cut!

**Reply by cortinico** [reply]
This is already included in RC0 ✅

**Comment by NickGerleman** [comment]
We should pick https://github.com/facebook/react-native/commit/e33767a36daf7901cb46854b6612dd52b0dbdeec to workaround a commonly reported Babel misconfiguration.

**Reply by cortinico** [reply]
Picked ✅

**Comment by cortinico** [comment]
Fast-refresh is broken on Bridgeless mode on Android. Here the fix:
- https://github.com/facebook/react-native/pull/43174

**Reply by cortinico** [reply]
Picked ✅

**Comment by cortinico** [comment]
Removing those files to clean the Upgrade Helper diff for 0.74:
- https://github.com/facebook/react-native/pull/43175

**Reply by cortinico** [reply]
Picked ✅

**Comment by cortinico** [comment]
Let's pick:
- https://github.com/facebook/react-native/pull/43127

From @byCedric as this is part of the implementation of https://github.com/react-native-community/discussions-and-proposals/pull/759

**Reply by cortinico** [reply]
Picked ✅

**Comment by dmytrorykun** [comment]
Please pick this https://github.com/facebook/react-native/pull/43210 to address the iOS codegen issue described here https://github.com/facebook/react-native/issues/43204

**Reply by huntie** [reply]
Picked ✅

**Comment by mikehardy** [comment]
Hi there! I found a macCatalyst CLI issue and posted a PR that works for me in my react-native-firebase + react-native-RC testing script https://github.com/react-native-community/cli/pull/2312

**Reply by huntie** [reply]
This hasn't landed in `main` yet, will be RC3.

**Reply by cortinico** [reply]
This hasn't made to a release of the CLI yet, will land in RC4

**Comment by pierroo** [comment]
Please pick this facebook/react-native#36766

It's been a while that InteractionManager.runAfterInteractions() is broken (at least since 0.72.x) and it is having dramatic effects on core components, including virtualized lists

a dev in there proposed a solution that did fix all my problems with it (although not sure it's viable)

**Reply by huntie** [reply]
This is not a pull request, so we can't pick this right now. We can review for RC3.

**Reply by huntie** [reply]
@pierroo There remains nothing to pick for RC3, please reply if there is more progress next week.

**Comment by cipolleschi** [comment]
Can we pick
* https://github.com/facebook/react-native/commit/c3b0a8f1626939cf5c7b3864a5acf9d3dad26fb3

This is not technically a fix, but it is a rename of a couple of APIs. In order not to make it breaking, we are first deprecating them in 0.74 to remove the old, not iOS-compliant names in 0.75.

**Reply by huntie** [reply]
Picked ✅

**Comment by cipolleschi** [comment]
Can we pick: 
* https://github.com/facebook/react-native/commit/a1197695fc7dd7750001dc60853fd19f162dc69e

this repositions the call to a warning to a better place to make sure that it works both in Bridgeless and Bridge mode.

**Reply by huntie** [reply]
Picked ✅

**Comment by cipolleschi** [comment]
Please, merge https://github.com/facebook/react-native/pull/43245
This solve a possible ENV var clash between a RN env var and other env vars users can set in their system.

**Reply by huntie** [reply]
Merged ✅

**Comment by cortinico** [comment]
Let's pick:
- https://github.com/facebook/react-native/pull/43219

This is currently causing some libraries that were invoking `TurboModule.class.isAssignableFrom` to crash at runtime (reported by @tido64)

**Reply by huntie** [reply]
Picked ✅

**Comment by RSNara** [comment]
Can we pull the following diffs into RC:

- Pulling PopupMenuAndroid out of React Native core: https://github.com/facebook/react-native/commit/8bced4b29db9b41a68ef726df81449a399eff604?fbclid=IwAR07o_l90X64Hl0eu2bVbGIiSUMhdU-sNuBjDDeYCpMdQLnnbw_32Gt54jM

- Fix for the extraction above: https://github.com/facebook/react-native/commit/fcabb2ad21b2c17f4e0d40b0637c22ba84568038?fbclid=IwAR3k75JV3DeFAYwSA5GHFHsQZ3rLyhBUB5-ItREKjz576trOtJ_cpdaa6N4

**Reply by huntie** [reply]
Picked ✅

**Comment by tomekzaw** [comment]
Can we please pick these two PRs (once merged) into RC?
* https://github.com/facebook/react-native/pull/43274
* https://github.com/facebook/react-native/pull/43275

Without them it's impossible to include header files and link shared libraries related to `Text` and `TextInput` components on Android.

**Reply by huntie** [reply]
Picked ✅

**Comment by javache** [comment]
Can we merge https://github.com/facebook/react-native/commit/5f45700bd06fb253650c8653b511b084b99eeb7f ? This prevents certain exceptions from being masked on Android bridgeless.

**Reply by huntie** [reply]
Picked ✅

**Comment by GijsWeterings** [comment]
(Copying from the top post, in case people scroll down immediately):

Now locked -> please submit pick requests and issue reports via https://github.com/reactwg/react-native-releases/issues/new/choose

All pick requests should now be filed through the [issue template](https://github.com/reactwg/react-native-releases/issues/new?assignees=&labels=Type%3A+Pick+Request&projects=&template=pick_request_form.yml)

You can see outstanding pick requests for 0.74 in its [dedicated project board](https://github.com/orgs/reactwg/projects/5/views/3)

See https://github.com/reactwg/react-native-releases/discussions/117 for more details

We will lock this discussion for now.
